packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,329 @@
+
+    protected async initializingFirstTime() {
+        const sharedMatrix = SharedMatrix.create(this.runtime, this.matrixKey);
+        this.root.set(this.matrixKey, sharedMatrix.handle);
+
+       const counter = SharedCounter.create(this.runtime, this.counterKey);
+       this.root.set(this.counterKey, counter.handle);
+
+       const dsFactory2 = await requestFluidObject<TestDataObject2>(
+       await this._context.containerRuntime.createDataStore(TestDataObjectType2), "");
+       this.root.set("dsFactory2", dsFactory2.handle);
+    }
+
+    protected async hasInitialized() {
+        const matrixHandle = this.root.get<IFluidHandle<SharedMatrix>>(this.matrixKey);
+        assert(matrixHandle !== undefined, "SharedMatrix not found");
+        this.matrix = await matrixHandle.get();
+
+        this.undoRedoStackManager = new UndoRedoStackManager();
+        this.matrix.insertRows(0, 3);
+        this.matrix.insertCols(0, 3);
+        this.matrix.openUndo(this.undoRedoStackManager);
+
+        const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(this.counterKey);
+        assert(counterHandle);
+        this.counter = await counterHandle.get();
+    }
+}
+const dataStoreFactory1 = new DataObjectFactory(
+    TestDataObjectType1,
+    TestDataObject1,
+    [SharedMap.getFactory(), SharedMatrix.getFactory(), SharedCounter.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const dataStoreFactory2 = new DataObjectFactory(
+    TestDataObjectType2,
+    TestDataObject2,
+    [SharedMap.getFactory(), SharedMatrix.getFactory(), SharedCounter.getFactory()],
+    [],
+    [],
+    createDataStoreRuntime(),
+);
+const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+    runtime.IFluidHandleContext.resolveHandle(request);
+
+const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+    [dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+    [dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+    ],
+);
+
+const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+    dataStoreFactory1,
+    registryStoreEntries,
+    undefined,
+    [innerRequestHandler],
+    runtimeOptions,
+);
+
+async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+): Promise<ISummarizer> {
+    return createSummarizerFromFactory(
+        provider,
+        container,
+        dataStoreFactory1,
+        summaryVersion,
+        getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore,
+        registryStoreEntries,
+    );
+}
+
+function createDataStoreRuntime(factory: typeof FluidDataStoreRuntime = FluidDataStoreRuntime) {
+    return mixinSummaryHandler(async (runtime: FluidDataStoreRuntime) => {
+        const obj: Provider<SearchContent> = (await DataObject.getDataObject(runtime)) as Provider<SearchContent>;
+        const searchObj = obj.SearchContent;
```

Comment:
All this search related code is not needed. All you need is to call `await DataObject.getDataObject(runtime)` and return undefined.
In the `hasInitialized` function of TestDataObject1, you need to request TestDataObject2. That should repro the issue.

Reply:
My goal was to be as close as an actual consumer would use it.

Reply:
That sounds fine. It just looks complex and confusing atm :). Maybe remove the types and interfaces and keep the other logic around? Or, add comments explaining what these are and why this test is testing them.

Reply:
I only got 100% repro using the search - using on that hasInitialized it did not repro 100% of the time.